packages/block-library/src/list/test/edit.native.js

```diff
@@ -347,10 +347,7 @@ describe( 'List block', () => {
 		// backward delete
 		const listItemField =
 			within( listItemBlock ).getByLabelText( /Text input. .*Two.*/ );
-		changeAndSelectTextOfRichText( listItemField, 'Two', {
-			initialSelectionStart: 0,
-			initialSelectionEnd: 3,
-		} );
+		selectTextInRichText( listItemField );
```

Comment:
This specific example relies upon `selectTextInRichText`'s default parameters of `selectionStart = 0` and `selectionEnd = selectionStart`. It is convenient to not need to pass `0` here to set the cursor to the beginning of the input. 

That said, does the name `selectTextInRichText` too strongly imply that (all) text will be selected by default? We cannot select _all_ text by default, as the helper is not receiving or know the string length. However, we could make `selectionStart` (and possibly `selectionEnd`) required to avoid any ambiguity of how this helper functions. 

What is better for this helper: convenience or clarity?

Reply:
> That said, does the name `selectTextInRichText` too strongly imply that (all) text will be selected by default? We cannot select _all_ text by default, as the helper is not receiving or know the string length.

In `typeInRichText` function we're getting the text length by checking the `value` prop of `RichText`, so maybe we could do the same for selection.

> However, we could make `selectionStart` (and possibly `selectionEnd`) required to avoid any ambiguity of how this helper functions.

Yep, when I read `selectTextInRichText` my first impression was that it will select all text. I was thinking about naming it differently like `moveCaretInRichtText`, but not sure if the name implies text selection. As you mentioned, probably making the argument `selectionStart` required would help to improve clarity. Regarding `selectionEnd`, not sure if I'd also make it required. 

> What is better for this helper: convenience or clarity?

Good question. I'd advocate for clarity in the spirit of making the tests easier to read. WDYT?

Reply:
> In `typeInRichText` function we're getting the text length by checking the `value` prop of `RichText`, so maybe we could do the same for selection.

Correct, but the `value` is unavailable within the `selectTextInRichText` function scope. We would have to add a special parameter to `typeInRichText` allowing communication of a "`typeInRichText` select the entire value (that only you know about) for me" intent. I'm less inclined expand the API surface of `typeInRichText` for this one-off, hidden (internal to the `selectTextInRichText` helper) use case. 

```js
function selectTextInRichText() {
  typeInRichText(richText, "", { selectAll: true });
}
```